src/components/NewCreditCardPageContent.tsx

```diff
@@ -197,9 +197,11 @@ export const NewCreditCardPageContent = () => {
         <Text weight="bold">다음</Text>
       </NextButton>
 
-      {isModalOpen && (
-        <CardCompanyListModal closeModal={closeModal} handleOnClick={handleCardCompanyChange} />
-      )}
+      <CardCompanyListModal
+        isModalOpen={isModalOpen}
+        closeModal={closeModal}
```

Comment:
```tsx
import type { PropsWithChildren } from "react";
import { useEffect } from "react";
import ReactDOM from "react-dom";
import styled from "styled-components";

const BackDrop = styled.div`
  position: fixed;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;

  background: rgba(0, 0, 0, 0.35);
`;

const Container = styled.div`
  position: fixed;
  bottom: 0;
  width: 100%;

  border-radius: 8px 8px 0 0;
  background: #ffffff;
`;

type ModalProps = {
  isOpen: boolean;
  closeModal: () => void;
  portalNode?: Element | null;
};

const Modal = (props: PropsWithChildren<ModalProps>) => {
  const { isOpen, closeModal, children, portalNode } = props;

  const handleKeyDown = (event: KeyboardEvent) => {
    if (event.key === "Escape") {
      closeModal();
    }
  };

  useEffect(() => {
    document.addEventListener("keydown", handleKeyDown);

    return () => {
      document.removeEventListener("keydown", handleKeyDown);
    };
  });

  const ModalContent = (
    <BackDrop onClick={closeModal}>
      <Container>{children}</Container>
    </BackDrop>
  );

  if (!isOpen) return null;

  if (portalNode) return ReactDOM.createPortal(ModalContent, portalNode);

  return ModalContent;
};

export default Modal;

```
제가 배포한 Modal 로직은 이렇습니다! isOpen에 따른 조건부 렌더링을 하기때문에 isOpen이 true일때 모달창이 띄어질 수 밖에없습니다! 그렇기때문에 closeModal props로 넘겨주는 함수에서 isOpen을 false로 바꿔주는 로직이 없다면 모달을 닫을 수가 없다고 생각합니다. 그래서 저 조건은 필수적이란 생각을 하는데요! 뭔가 설명이 부족한거 같아서 README에 좀 더 명확한 설명을 추가해주었습니다.

isOpen 설명에 `This modal library displays the modal when isOpen is true, and does not display it when it is false` 설명을 추가했고, closeModal에 `To close the modal in this modal library, isOpen needs to be set to false since the modal is displayed when isOpen is true. So, Should contain the logic to change isOpen to false in this function.` 이 설명을 추가해주었습니다!

Reply:
처음에 페이지 라우팅 시 모달을 무조건 띄우도록 만들었었고, esc 키 입력 시 모달을 닫는 기능도 없었어서 카드사 클릭할 때만 모달이 닫히게 만들었어서 카드사 유효성 검사에 대해 생각을 못했네요! 카드사 유효성 검사를 추가해주었습니다! 또, 카드이름 유효성 검사도 추가해줬어요!

Reply:
https://github.com/yeopto/react-yt-modal/pull/1 😁